crates/node/src/main.rs

```diff
@@ -74,32 +71,12 @@ fn main() -> anyhow::Result<()> {
         .build()
         .unwrap()
         .block_on(async move {
-            enum FutureOutput {
-                CtrlC,
-                SigTerm,
-                Node,
-            }
-
             let shutdown_signal = cfg.shutdown_signal.clone();
-
-            let mut shutdown_fut =
-                pin!(tokio::signal::ctrl_c().map(|_| FutureOutput::CtrlC).fuse());
-
-            let mut sigterm = tokio::signal::unix::signal(SignalKind::terminate())?;
-            let mut sigterm_fut = pin!(sigterm.recv().map(|_| FutureOutput::SigTerm).fuse());
-
-            let mut node_fut = pin!(wcn_node::run(cfg).await?.map(|_| FutureOutput::Node));
-
-            loop {
-                match (&mut shutdown_fut, &mut sigterm_fut, &mut node_fut)
-                    .race()
-                    .await
-                {
-                    FutureOutput::CtrlC => shutdown_signal.emit(),
-                    FutureOutput::SigTerm => shutdown_signal.emit(),
-                    FutureOutput::Node => return Ok(()),
-                }
-            }
+            let run_fut = wcn_node::run(cfg)
+                .await?
+                .tap(|_| tracing::info!("node stopped"));
```

Comment:
It will just print the log right away. Not sure what was the original intention.